src/main/java/dinkplugin/notifiers/PetNotifier.java

```diff
@@ -22,18 +37,52 @@ protected String getWebhookUrl() {
     }
 
     public void onChatMessage(String chatMessage) {
-        if (isEnabled() && PET_REGEX.matcher(chatMessage).matches()) {
-            this.handleNotify();
+        if (petName == null && isEnabled() && PET_REGEX.matcher(chatMessage).matches()) {
+            this.petName = "";
         }
     }
 
+    public void onClanNotification(String message) {
+        if (petName != null) {
```

Comment:
Could we change the initial null-check to an early out instead? also might want to check for `petName == ""` (or inverse if we early-out) instead, since that's what we're actually looking for (i.e. Has the pet notifier been primed with a pet notification message?)